cmd/gardener-metrics-exporter/app.go

```diff
@@ -122,30 +124,45 @@ func run(o *options, closeCh chan os.Signal) error {
 	return nil
 }
 
+func newConfigFromBytes(kubeconfig string) (*restclient.Config, error) {
+	kubecf, err := ioutil.ReadFile(kubeconfig)
+	if err != nil {
+		log.Errorf("Could not read config")
+	}
+	configObj, err := clientcmd.Load(kubecf)
+	if err != nil {
+		log.Errorf("Could not load config")
+	}
+	clientConfig := clientcmd.NewDefaultClientConfig(*configObj, &clientcmd.ConfigOverrides{})
+	client, err := clientConfig.ClientConfig()
+	if err != nil {
+		return nil, nil
+	}
+	if client == nil {
+		return nil, err
```

Comment:
Here we should create an error object and return it because err can be nil in this case.
`return nil, errors.new("for example: client is nil")`